File: app/src/main/java/social/entourage/android/api/tape/EncounterTaskResult.java
package social.entourage.android.api.tape;

import social.entourage.android.api.model.map.Encounter;

public class EncounterTaskResult {

    public enum OperationType {
        ENCOUNTER_ADD,
        ENCOUNTER_UPDATE
    };

    private boolean success;
    private Encounter encounter;
    private OperationType operationType;

    public EncounterTaskResult(boolean success, Encounter encounter, OperationType operationType) {
        this.success = success;
        this.encounter = encounter;
        this.operationType = operationType;
    }

    public boolean isSuccess() {
        return success;
    }

    public Encounter getEncounter() {
        return encounter;
    }

    public OperationType getOperationType() {
        return operationType;
    }
}

File: app/src/main/java/social/entourage/android/api/tape/EncounterTaskResult.kt
package social.entourage.android.api.tape

import social.entourage.android.api.model.tour.Encounter

class EncounterTaskResult(val isSuccess: Boolean, val encounter: Encounter, val operationType: OperationType) {
    enum class OperationType {
        ENCOUNTER_ADD, ENCOUNTER_UPDATE
    }

}